src/azure_devtools/scenario_tests/preparers.py

```diff
@@ -22,6 +27,7 @@ def __init__(self, name_prefix, name_len, disable_recording=False):
         self.test_class_instance = None
         self.live_test = False
         self.disable_recording = disable_recording
+        self.no_cleanup_on_fail = os.environ.get(ENV_TEST_NO_CLEANUP_ON_FAIL, None)
```

Comment:
You should input this property from __init__ functino signature instead of read the value from environment in __init__